packages/big-design/src/components/Worksheet/hooks/spec.tsx

```diff
@@ -0,0 +1,91 @@
+import { renderHook } from '@testing-library/react-hooks';
+import React from 'react';
+import { act } from 'react-test-renderer';
+
+import { Cell } from '../types';
+
+import { useEditableCell } from './useEditableCell';
+
+let mockUpdateItems = jest.fn();
+jest.mock('./useUpdateItems', () => ({
+  useUpdateItems: () => ({
+    updateItems: mockUpdateItems,
+  }),
+}));
+
+describe('useEditableCells', () => {
+  const cell: Cell<unknown> = { rowIndex: 0, columnIndex: 0, type: 'text', hash: 'category', value: 'Text' };
+
+  let preventDefault: () => void;
+  let stopPropagation: () => void;
+
+  beforeEach(() => {
+    mockUpdateItems = jest.fn();
+    preventDefault = jest.fn();
+    stopPropagation = jest.fn();
+  });
+
+  test('double click', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => result.current.handleDoubleClick());
+
+    expect(result.current.isEditing).toBe(true);
+  });
+
+  test('blur', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleDoubleClick();
+      result.current.handleBlur();
+    });
+
+    expect(result.current.isEditing).toBe(false);
+  });
+
+  test('enter', () => {
+    const { result } = renderHook(() => useEditableCell(cell));
+
+    act(() => {
+      result.current.handleDoubleClick();
+      result.current.handleKeyDown(
+        ({
+          currentTarget: 1,
```

Comment:
curious, what are these `currentTarget` and `target` fields for in a React.KeyboardEvent? Can we just not supply them?

Also, can we not construct this object using the native API `new KeyboardEvent('Foo', { key: 'Enter' })` ?

Reply:
To be honest it's a copy pasta from another test. I can simply remove them since they are not relevant to current test, good catch. I couldn't implement your solution, can you expand on it?